search-cache-pipeline.yml

```diff
@@ -0,0 +1,83 @@
+schedules:
+- cron: 0 * * * *
+  displayName: Run every hour
+  branches:
+    include:
+    - main
+  always: true
+
+pool:
+    name: Hosted Ubuntu 1604
+
+steps:
+- checkout: self
+
+- script: '$(Build.SourcesDirectory)/build.sh'
+
+- task: UseDotNet@2
+  displayName: 'Use .NET 3.1'
+  inputs:
+    packageType: sdk
+    version: 3.1.x
+    installationPath: $(Build.SourcesDirectory)/.dotnet
+
+- task: UseDotNet@2
+  displayName: 'Use .NET 5.0.100'
+  inputs:
+    packageType: sdk
+    version: 5.0.100
+    installationPath: $(Build.SourcesDirectory)/.dotnet
+
+- task: UseDotNet@2
+  displayName: 'Use .NET 5.0.300'
+  inputs:
+    packageType: sdk
+    version: 5.0.300
+    installationPath: $(Build.SourcesDirectory)/.dotnet
+
+- bash: >
+    curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin
+    -Version 6.0.100-rc.2.21420.30
+    -InstallDir $(Build.SourcesDirectory)/.dotnet
+    -SkipNonVersionedFiles
+  displayName: Install .NET 6.0
+
+- bash: >
+    $(Build.SourcesDirectory)/.dotnet/dotnet $(Build.SourcesDirectory)/artifacts/bin/Microsoft.TemplateSearch.TemplateDiscovery/Debug/net6.0/Microsoft.TemplateSearch.TemplateDiscovery.dll
+    --basePath $(System.DefaultWorkingDirectory)/NugetDownloadDirectory --allowPreviewPacks -v --test
+  displayName: Run Cache Updater
+
+- task: CopyFiles@2
```

Comment:
why do we need this task?

Reply:
Publish task does not support wildcards. We cannot selectively publish only the .json files, it will publish everything under that directory. I'm copying the json's to isolate them in a single folder.